### PR TITLE
Feature/global style

### PR DESCRIPTION
**Pull request by kobizz**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/kits/assets/js/component.js [hunk 1]
```diff
@@ -15,11 +15,11 @@ export default class extends $e.modules.ComponentBase {
 				icon: 'eicon-site-identity',
 				helpUrl: 'http://go.elementor.com/panel-site-identity',
 			},
-			// 'colors-and-typography': {
-			// 	title: elementor.translate( 'Colors & Typography' ),
-			// 	icon: 'eicon-colors-typography',
-			// 	helpUrl: 'http://go.elementor.com/panel-colors-and-typography',
-			// },
+			'global-style': {
```
Comment (matipojo) [inline core/kits/assets/js/component.js:18]:
```suggestion
			'colors-and-typography': {
```

File: core/kits/assets/js/component.js [hunk 2]
```diff
@@ -15,11 +15,11 @@ export default class extends $e.modules.ComponentBase {
...
+				title: elementor.translate( 'Colors & Typography' ),
+				icon: 'eicon-colors-typography',
+				helpUrl: 'http://go.elementor.com/panel-global-style',
```
Comment (matipojo) [inline core/kits/assets/js/component.js:21]:
```suggestion
				helpUrl: 'http://go.elementor.com/panel-colors-and-typography',
```

File: is-it-elementor.html [hunk 3]
```diff
@@ -0,0 +1,111 @@
+<!DOCTYPE html>
```
Comment (matipojo) [inline is-it-elementor.html:1]: ?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -97,6 +97,8 @@
 # Modules
 /modules                          		@matipojo
 /modules/dynamic-tags             		@matipojo @kobizz
+/modules/library						@matipojo @kobizz
+/modules/system-info					@matipojo @kobizz
```
[the remark this answers is not in the document]
Comment (kobizz) [reply .github/CODEOWNERS]: You reviewed an old commit

File: assets/dev/js/editor/editor-base.js [hunk 5]
```diff
@@ -759,7 +753,7 @@ export default class EditorBase extends Marionette.Application {
 				this.addWidgetsCache( data );
 
 				if ( this.loaded ) {
-					this.schemes.printSchemesStyle();
+					// this.schemes.printSchemesStyle();
```
[the remark this answers is not in the document]
Comment (kobizz) [reply assets/dev/js/editor/editor-base.js:756]: It's waiting for our discussion

File: core/kits/documents/kit.php [hunk 6]
```diff
@@ -21,7 +21,7 @@ public function __construct( array $data = [] ) {
 		$this->tabs = [
 			'site_identity' => new Tabs\Site_Identity( $this ),
 			'lightbox' => new Tabs\Lightbox( $this ),
-			'colors_and_typography' => new Tabs\Colors_And_Typography( $this ),
+			'global_style' => new Tabs\Global_Style( $this ),
```
Comment (matipojo) [inline core/kits/documents/kit.php:24]: The panel page title is "Colors and Typography" so, Please rename all related code to this term.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by rodolphebertozzo — DISMISSED** [review]

File: core/kits/assets/js/component.js [hunk 7]
```diff
@@ -31,7 +31,7 @@ export default class extends $e.modules.ComponentBase {
 				helpUrl: 'http://go.elementor.com/panel-layout-settings',
 			},
 			'theme-style': {
-				title: elementor.translate( 'Theme Style' ),
+				title: elementor.translate( 'theme_style' ),
```
Comment (matipojo) [inline core/kits/assets/js/component.js:34]: Revert like on lines 19, 24.

File: core/kits/documents/tabs/colors-and-typography.php [hunk 8]
```diff
@@ -35,23 +36,129 @@ public function register_tab_controls() {
...
+
+		$repeater->add_group_control(
+			Group_Control_Typography::get_type(),
+			[
+				'name' => 'styles',
+				'label' => '',
+				'fields_options' => [
+					'font_family' => [
+						'selectors' => [
+							'{{SELECTOR}}' => '--e-global-style-{{external._id.VALUE}}-font-family: "{{VALUE}}"',
```
Comment (matipojo) [inline core/kits/documents/tabs/colors-and-typography.php:103]:
```suggestion
							'{{SELECTOR}}' => '--e-global-style-{{external:_id}}-font-family: "{{VALUE}}"',
```
```suggestion
							'{{SELECTOR}}' => '--e-global-style-{{external-value:_id}}-font-family: "{{VALUE}}"',
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

File: assets/dev/js/editor/utils/controls-css-parser.js [hunk 9]
```diff
@@ -39,35 +39,140 @@ ControlsCSSParser = elementorModules.ViewModule.extend( {
...
-			placeholders,
-			replacements
-		);
+		var value = this.getStyleControlValue( control, values );
+
+		if ( undefined === value ) {
+			return;
+		}
+
+		_.each( control.selectors, ( cssProperty, selector ) => {
```
Comment (matipojo) [inline assets/dev/js/editor/utils/controls-css-parser.js:68]: It doesn't work for `multiple` based controls like font-size.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/utils/controls-css-parser.js (hunk 9 above)
Comment (kobizz) [reply assets/dev/js/editor/utils/controls-css-parser.js:68]: It works fine for me

File: assets/dev/js/editor/utils/controls-css-parser.js [hunk 10]
```diff
@@ -39,35 +39,179 @@ ControlsCSSParser = elementorModules.ViewModule.extend( {
 	},
 
 	addStyleRules: function( styleControls, values, controls, placeholders, replacements ) {
-		var self = this,
-			dynamicParsedValues = self.getSettings( 'settingsModel' ).parseDynamicSettings( values, self.getSettings( 'dynamicParsing' ), styleControls );
+		// If the current element contains dynamic values, parse thesex values
```
Comment (matipojo) [inline assets/dev/js/editor/utils/controls-css-parser.js:42]:
```suggestion
		// If the current element contains dynamic values, parse these values
```

File: assets/dev/js/editor/utils/controls-css-parser.js [hunk 11]
```diff
@@ -39,35 +39,179 @@ ControlsCSSParser = elementorModules.ViewModule.extend( {
...
-		_.each( styleControls, function( control ) {
+		_.each( styleControls, ( control ) => {
 			if ( control.styleFields && control.styleFields.length ) {
-				self.addRepeaterControlsStyleRules( values[ control.name ], control.styleFields, controls, placeholders, replacements );
+				this.addRepeaterControlsStyleRules( values[ control.name ], control.styleFields, control.fields, placeholders, replacements );
 			}
 
-			if ( control.dynamic && control.dynamic.active && values.__dynamic__ && values.__dynamic__[ control.name ] ) {
-				self.addDynamicControlStyleRules( values.__dynamic__[ control.name ], control );
+			// If a dynamic tag includes controls with CSS implementations, Take their CSS and apply it
```
Comment (matipojo) [inline assets/dev/js/editor/utils/controls-css-parser.js:50]:
```suggestion
			// If a dynamic tag includes controls with CSS implementations, Take their CSS and apply it.
```

File: core/kits/documents/tabs/colors-and-typography.php [hunk 12]
```diff
@@ -35,23 +38,173 @@ public function register_tab_controls() {
...
+			]
+		);
+
+		$this->add_control(
+			'system_colors',
+			[
+				'type' => Global_Style_Repeater::CONTROL_TYPE,
+				'fields' => $repeater->get_controls(),
+				'default' => [
+					[
```
Comment (matipojo) [inline core/kits/documents/tabs/colors-and-typography.php:63]: Add an `_id` for each row.

File: core/kits/documents/tabs/colors-and-typography.php [hunk 13]
```diff
@@ -35,23 +38,173 @@ public function register_tab_controls() {
...
+
+		$font_weight_key = self::TYPOGRAPHY_GROUP_PREFIX . '_font_weight';
+
+		$this->add_control(
+			'system_typography',
+			[
+				'type' => Global_Style_Repeater::CONTROL_TYPE,
+				'fields' => $repeater->get_controls(),
+				'default' => [
+					[
```
Comment (matipojo) [inline core/kits/documents/tabs/colors-and-typography.php:175]: Add an `_id` for each row.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/kits/documents/tabs/colors-and-typography.php [hunk 14]
```diff
@@ -35,23 +38,181 @@ public function register_tab_controls() {
...
```
Comment (matipojo) [inline core/kits/documents/tabs/colors-and-typography.php:179]: Check `elementor_disable_typography_schemes` before adding defaults.

File: core/kits/documents/tabs/colors-and-typography.php [hunk 15]
```diff
@@ -35,23 +38,181 @@ public function register_tab_controls() {
...
+				],
+			]
+		);
+
+		$this->add_control(
+			'system_colors',
+			[
+				'type' => Global_Style_Repeater::CONTROL_TYPE,
+				'fields' => $repeater->get_controls(),
+				'default' => [
```
Comment (matipojo) [inline core/kits/documents/tabs/colors-and-typography.php:62]: Check `elementor_disable_color_schemes` before adding defaults.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/kits/documents/tabs/colors-and-typography.php [hunk 16]
```diff
@@ -35,23 +39,193 @@ public function register_tab_controls() {
...
 				'dynamic' => [],
+				'selectors' => [
+					'{{WRAPPER}}' => '--e-global-color-{{_id.VALUE}}: {{VALUE}}',
+				],
+			]
+		);
+
+		$default_colors = [];
+
+		if ( Plugin::$instance->kits_manager->is_custom_colors_enabled() ) {
```
Comment (matipojo) [inline core/kits/documents/tabs/colors-and-typography.php:60]: Due to the new terminology - can you rename the two methods to `is_system_colors_enabled`, `is_system_typography_enabled`.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**